File: bench/live_suite/sieve.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int run_sieve(int limit) {
    char *is_prime = (char *)malloc(limit + 1);
    memset(is_prime, 1, limit + 1);
    is_prime[0] = 0;
    is_prime[1] = 0;

    for (int p = 2; p * p <= limit; p++) {
        if (is_prime[p]) {
            for (int k = p * p; k <= limit; k += p) {
                is_prime[k] = 0;
            }
        }
    }

    int count = 0;
    for (int c = 0; c <= limit; c++) {
        if (is_prime[c]) count++;
    }
    free(is_prime);
    return count;
}

int main(void) {
    int total_primes = 0;
    for (int iter = 0; iter < 10; iter++) {
        total_primes = run_sieve(1000000);
    }
    printf("Primes count: %d\n", total_primes);
    return 0;
}

File: bench/live_suite/sieve.go
package main

import "fmt"

func runSieve(limit int) int {
	isPrime := make([]bool, limit+1)
	for i := 2; i <= limit; i++ {
		isPrime[i] = true
	}

	for p := 2; p*p <= limit; p++ {
		if isPrime[p] {
			for k := p * p; k <= limit; k += p {
				isPrime[k] = false
			}
		}
	}

	count := 0
	for c := 2; c <= limit; c++ {
		if isPrime[c] {
			count++
		}
	}
	return count
}

func main() {
	totalPrimes := 0
	for iter := 0; iter < 10; iter++ {
		totalPrimes = runSieve(1000000)
	}
	fmt.Printf("Primes count: %d\n", totalPrimes)
}
